test/unittests/libtesteth/testHelperTest.cpp

```diff
@@ -104,5 +105,98 @@ BOOST_AUTO_TEST_CASE(translateNetworks_leFrontier)
     }
 }
+BOOST_AUTO_TEST_SUITE_END()
+
+BOOST_AUTO_TEST_SUITE(TestHelper)
+BOOST_AUTO_TEST_CASE(levenshteinDistance_similar)
+{
+    char const* word1 = "someword";
+    char const* word2 = "soemword";
+    size_t distance = test::levenshteinDistance(word1, strlen(word1), word2, strlen(word2));
+    BOOST_CHECK_EQUAL(distance, 2);
+}
+
+BOOST_AUTO_TEST_CASE(levenshteinDistance_similar2)
+{
+    char const* word1 = "sOmeWord";
+    char const* word2 = "someword";
+    size_t distance = test::levenshteinDistance(word1, strlen(word1), word2, strlen(word2));
+    BOOST_CHECK_EQUAL(distance, 2);
+}
+
+BOOST_AUTO_TEST_CASE(levenshteinDistance_similar3)
+{
+    char const* word1 = "sOmeWoRd";
+    char const* word2 = "someword";
+    size_t distance = test::levenshteinDistance(word1, strlen(word1), word2, strlen(word2));
+    BOOST_CHECK_EQUAL(distance, 3);
+}
+
+BOOST_AUTO_TEST_CASE(levenshteinDistance_similar4)
+{
+    char const* word1 = "sOmeWoRd";
+    char const* word2 = "soemword";
+    size_t distance = test::levenshteinDistance(word1, strlen(word1), word2, strlen(word2));
+    BOOST_CHECK_EQUAL(distance, 5);
+}
+
+BOOST_AUTO_TEST_CASE(levenshteinDistance_AgtB)
+{
+    char const* word1 = "someword";
+    char const* word2 = "other";
+    size_t distance = test::levenshteinDistance(word1, strlen(word1), word2, strlen(word2));
+    BOOST_CHECK_EQUAL(distance, 4);
+}
+
+BOOST_AUTO_TEST_CASE(levenshteinDistance_AgtB2)
+{
+    char const* word1 = "some long sentence here";
+    char const* word2 = "other shorter phrase";
+    size_t distance = test::levenshteinDistance(word1, strlen(word1), word2, strlen(word2));
+    BOOST_CHECK_EQUAL(distance, 14);
+}
+
+BOOST_AUTO_TEST_CASE(levenshteinDistance_BgtA)
+{
+    char const* word1 = "other";
+    char const* word2 = "someword";
+    size_t distance = test::levenshteinDistance(word1, strlen(word1), word2, strlen(word2));
+    BOOST_CHECK_EQUAL(distance, 4);
+}
+
+BOOST_AUTO_TEST_CASE(levenshteinDistance_BgtA2)
+{
+    char const* word1 = "other shorter phrase";
+    char const* word2 = "some long sentence here";
+    size_t distance = test::levenshteinDistance(word1, strlen(word1), word2, strlen(word2));
+    BOOST_CHECK_EQUAL(distance, 14);
+}
+
+BOOST_AUTO_TEST_CASE(levenshteinDistance_different)
+{
+    char const* word1 = "abdefg";
+    char const* word2 = "hijklmn";
+    size_t distance = test::levenshteinDistance(word1, strlen(word1), word2, strlen(word2));
+    BOOST_CHECK_EQUAL(distance, 6);
+}
+
+BOOST_AUTO_TEST_CASE(getTestSuggestions)
+{
+    // this return might change by more new tests get added
+    auto list = test::getTestSuggestions("blocksuit");
+    BOOST_CHECK_EQUAL(list[0], "BlockSuite/ByzantiumBlockSuite");
+    BOOST_CHECK_EQUAL(list[1], "ClientBase/blocks");
+    BOOST_CHECK_EQUAL(list[2], "BlockSuite");
+}
+
+BOOST_AUTO_TEST_CASE(getTestSuggestions2)
+{
+    // this return might change by more new tests get added
```

Comment:
This is maintenance pain. Instead of using `c_allTestNames` in `getTestSuggestions` directly, you could pass it as a parameter, then pass the fake "all tests" here in tests.

Reply:
I would actually like to test that cmake constructed a valid test list array. 
stExample2. Must find and suggest stExample test. I am planning to make it irrelevant with the order. Which somehow is different on the build